src/components/dls/EmbeddableContent/ContentModal.tsx

```diff
@@ -0,0 +1,55 @@
+import { useEffect } from 'react';
+
+import * as Dialog from '@radix-ui/react-dialog';
+import { useRouter } from 'next/router';
+
+import CloseIcon from '../../../../public/icons/close.svg';
+import Button, { ButtonShape, ButtonVariant } from '../Button/Button';
+
+import styles from './ContentModal.module.scss';
+
+import { fakeNavigate } from 'src/utils/navigation';
+
+type ContentModalProps = {
+  isOpen?: boolean;
+  onClose?: () => void;
+  children: React.ReactNode;
+  hasCloseButton?: boolean;
+  url?: string;
+};
+
+const ContentModal = ({ isOpen, onClose, hasCloseButton, children, url }: ContentModalProps) => {
+  const router = useRouter();
+
+  useEffect(() => {
+    if (isOpen) fakeNavigate(url);
+    else fakeNavigate(router.asPath);
+
+    return () => {
+      fakeNavigate(router.asPath);
+    };
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
why the eslint ignore here?

Reply:
We only want to run the effect when the modal is is just opened and closed.

So we don't want to use `url` as deps. 

here's the scenario.
1. The modal is opened -> the url changed to `/2/1/tafsirs?tafsirId=90`
2. We change the verse number and the chapterId -> the url becomes `/4/5/tafsirs?tafsirId=90`
3. We update the tafsirId to `169` -> this update the redux of the tafsirId, so by the data flow this will update the `url` props.
  -> so the url now becomes `/2/1/tafsirs/tafsirId=90` again. Notice the `/2/1` is now reverted back to the initial value